wire-library/wire-runtime-swift/src/main/swift/CodingUserInfoKey+Wire.swift

```diff
@@ -52,61 +44,98 @@ extension JSONEncoder {
     }
 
     /// The encoding strategy to use for StringEncoded types that are themselves Encodable
-    /// Defaults to .string
+    /// - Note: Defaults to .string
     /// - Note: ProtoMap Dictionary keys are always encoded as strings
+    /// - SeeAlso: [Proto3 JSON Mapping](https://developers.google.com/protocol-buffers/docs/proto3#json)
     public enum StringEncodedEncodingStrategy {
         /// Encodes the string-encoded value, like `"myValue": "1"`
         case string
         /// Encodes the raw Encodable value, like `"myValue": 1`
         case raw
     }
+
+    /// The encoding strategy to use for key names in Codable implementations
+    /// - Note: Defaults to .camelCase
+    /// - SeeAlso: [Proto3 JSON Mapping](https://developers.google.com/protocol-buffers/docs/proto3#json)
+    public enum KeyNameEncodingStrategy {
+        // Convert key names to `camelCase`
+        case camelCase
+        // Maintain the original field names, typically `snake_case`
+        case fieldName
+    }
+
+    /// The encoding strategy to use for optional values and collections
+    /// - Note: Defaults to .skip
+    /// - SeeAlso: [Proto3 JSON Mapping](https://developers.google.com/protocol-buffers/docs/proto3#json)
+    public enum DefaultValuesEncodingStrategy {
+        // Skip "default" values
+        case skip
+        // Include "default" values
+        case include
```

Comment:
> A proto3 JSON implementation may provide the following options:
> 
> Emit fields with default values: Fields with default values are omitted by default in proto3 JSON output. An implementation may provide an option to override this behavior and output fields with their default values.

That's why I used `emit` before and why I have the name that I did.
Open to suggestions @efirestone

Reply:
Google's JVM library calls it "output" (`alwaysOutputDefaultValueFields`), which I find confusing. Some of the other Wire code uses emitter (`EmitterRules`), so I'm down with that. `include` probably makes for the nicest variable name (`includeDefaults`), so it seems good/fine too. I defer to you.

Reply:
OK I'll leave it as is